vapoursynth jitasm clang

vapoursynth seem to fail when compiled with the clang compilers, i get a rather obscure error about M32 haing 6 elements but only 5 given. Looking at the jitasm.h code does not really give me a clue as to what might cause this as the function seems to be overloaded quite a few times.

error here ->
```

../vapoursynth-R49/src/core/jitasm.h:8618:11: error: no matching constructor for initialization of 'jitasm::Mem32' (aka 'MemT<OpdT<32>>')
                                Mem32 lo(val_.GetAddressBaseSize(), val_.GetBase(), val_.GetIndex(), val_.GetScale(), val_.GetDisp());
                                      ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../vapoursynth-R49/src/core/jitasm.h:503:2: note: candidate constructor not viable: requires 6 arguments, but 5 were provided
        MemT(OpdSize base_size, OpdSize index_size, const RegID& base, const RegID& index, sint64 scale, sint64 disp) : OpdN(base_size, index_size, base, index, scale, disp) {}
        ^
../vapoursynth-R49/src/core/jitasm.h:501:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 5 were provided
struct MemT : OpdN
       ^
../vapoursynth-R49/src/core/jitasm.h:501:8: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 5 were provided
../vapoursynth-R49/src/core/jitasm.h:8619:11: error: no matching constructor for initialization of 'jitasm::Mem32' (aka 'MemT<OpdT<32>>')
                                Mem32 hi(val_.GetAddressBaseSize(), val_.GetBase(), val_.GetIndex(), val_.GetScale(), val_.GetDisp() + 4);
                                      ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../vapoursynth-R49/src/core/jitasm.h:503:2: note: candidate constructor not viable: requires 6 arguments, but 5 were provided
        MemT(OpdSize base_size, OpdSize index_size, const RegID& base, const RegID& index, sint64 scale, sint64 disp) : OpdN(base_size, index_size, base, index, scale, disp) {}
        ^
../vapoursynth-R49/src/core/jitasm.h:501:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 5 were provided
struct MemT : OpdN
       ^

```

--- a/mingw-w64-vapoursynth/0001-wrong-arguments-to-Mem32-constructor.patch
+++ b/mingw-w64-vapoursynth/0001-wrong-arguments-to-Mem32-constructor.patch
@@ -1,0 +1,13 @@
+--- a/src/core/expr/jitasm.h
++++ b/src/core/expr/jitasm.h
+@@ -8609,8 +8609,8 @@
+ #else
+ 			if (val_.IsMem()) {
+ 				// from memory
+-				Mem32 lo(val_.GetAddressBaseSize(), val_.GetBase(), val_.GetIndex(), val_.GetScale(), val_.GetDisp());
+-				Mem32 hi(val_.GetAddressBaseSize(), val_.GetBase(), val_.GetIndex(), val_.GetScale(), val_.GetDisp() + 4);
++				Mem32 lo(val_.GetAddressBaseSize(), val_.GetAddressIndexSize(), val_.GetBase(), val_.GetIndex(), val_.GetScale(), val_.GetDisp());
++				Mem32 hi(val_.GetAddressBaseSize(), val_.GetAddressIndexSize(), val_.GetBase(), val_.GetIndex(), val_.GetScale(), val_.GetDisp() + 4);
+ 				f.mov(f.eax, lo);
+ 				f.mov(f.edx, hi);
+ 			} else if (val_.IsImm()) {

--- a/mingw-w64-vapoursynth/PKGBUILD
+++ b/mingw-w64-vapoursynth/PKGBUILD
@@ -4,7 +4,7 @@ _realname=vapoursynth
 pkgbase=mingw-w64-${_realname}
 pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
 pkgver=57
-pkgrel=1
+pkgrel=2
 pkgdesc="A video processing framework with simplicity in mind (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32')
@@ -18,11 +18,14 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-gcc"
              "${MINGW_PACKAGE_PREFIX}-pkgconf"
              "${MINGW_PACKAGE_PREFIX}-python-sphinx")
 options=(!strip staticlibs)
-source=("vapoursynth-${pkgver}.tar.gz::https://github.com/vapoursynth/vapoursynth/archive/R${pkgver}.tar.gz")
-sha256sums=('9bed2ab1823050cfcbdbb1a57414e39507fd6c73f07ee4b5986fcbf0f6cb2d07')
+source=("vapoursynth-${pkgver}.tar.gz::https://github.com/vapoursynth/vapoursynth/archive/R${pkgver}.tar.gz"
+        "0001-wrong-arguments-to-Mem32-constructor.patch")
+sha256sums=('9bed2ab1823050cfcbdbb1a57414e39507fd6c73f07ee4b5986fcbf0f6cb2d07'
+            '085331d7c81da760b66864ec34647094384514689e7ea9895e7bd34ed9ce9831')
 
 prepare() {
   cd "${srcdir}/${_realname}-R${pkgver}"
+  patch -Nbp1 -i "${srcdir}/0001-wrong-arguments-to-Mem32-constructor.patch"
   autoreconf -vfi
 }
 

